packages/components/src/components/bal-slider/bal-slider-item/bal-slider-item.tsx

```diff
@@ -1,11 +1,19 @@
-import { Component, ComponentInterface, h, Host } from '@stencil/core'
+import { Component, ComponentInterface, h, Host, Method, Prop } from '@stencil/core'
 
 @Component({
   tag: 'bal-slider-item',
 })
 export class TabSliderItem implements ComponentInterface {
+  /**
+   * Label of the slide which will be used for pagination tabs
+   */
+  @Prop() label = ''
   private inputId = `bal-slider-item-${sliderItemId++}`
 
+  @Method() async getLabel(): Promise<string> {
```

Comment:
you do not need a method since there is a public prop available

Reply:
@hirsch88 yeah, but that prop is not needed inside of the component, so the compiler will say 'declared but not used' and I don't need to use it in there because the content of the slide item is different depending on the use case. What other solution do you see?

Reply:
@madaluca 

```typescript
  private updateSlides() {
    this.slides = this.getChildItems()
    this.slidesLabels = this.slides.map(slide => slide.label)
  }
```